### ISSUE TITLE
Roslaunch ERROR

### ISSUE THREAD

**Issue by Cxz-dev** [engcang/oakd-ros-simple#5]
Hi, thanks for your code!

       I got an error when using "main.launch".

![2023-01-06 10-16-37 的屏幕截图](https://user-images.githubusercontent.com/55902326/210916687-7d9eaab3-f8f4-4f50-95e4-36e867db5c5f.png)

       Then I set "get_imu=false", another error occured.

![2023-01-06 10-21-51 的屏幕截图](https://user-images.githubusercontent.com/55902326/210917041-8cbe2abb-4333-4f3d-aa65-0e2a7bff3dba.png)

      Can you give me some advice? Thank u!

**Comment by Cxz-dev** [comment]
Ubuntu18.04
Device: OAK-D-PRO-W

**Comment by Cxz-dev** [comment]
Ubuntu18.04
Device: OAK-D-PRO

**Comment by engcang** [comment]
Hi @Cxz-dev . Can you attach the GDB debugging log here?

Please change this line - https://github.com/engcang/oakd-ros-simple/blob/main/launch/main.launch#L53 
into
<node name="oakd_ros_node" pkg="oakd_ros" type="oakd_ros_node" output="screen" launch-prefix="gdb -ex run --args"/>

After you launch and get the error, 
then type "bt" and press enter, then you will see the GDB log about the error.

**Comment by Cxz-dev** [comment]
Hi @engcang . I try the GDB debugging log.

GNU gdb (Ubuntu 8.1.1-0ubuntu1) 8.1.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/cxz/dai_ws/devel/lib/oakd_ros/oakd_ros_node...done.
Starting program: /home/cxz/dai_ws/devel/lib/oakd_ros/oakd_ros_node __name:=oakd_ros_node __log:=/home/cxz/.ros/log/be435d44-92df-11ed-ad51-48e7daba6333/oakd_ros_node-1.log
[tcsetpgrp failed in terminal_inferior: 对设备不适当的 ioctl 操作]
[tcsetpgrp failed in terminal_inferior: 无效的参数]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[tcsetpgrp failed in terminal_inferior: 无效的参数]
[tcsetpgrp failed in terminal_inferior: 无效的参数]
[New Thread 0x7fffea1e8700 (LWP 19584)]
[New Thread 0x7fffe99e7700 (LWP 19585)]
[New Thread 0x7fffe91e6700 (LWP 19586)]
[New Thread 0x7fffe89e5700 (LWP 19587)]
[New Thread 0x7fffdbfff700 (LWP 19588)]
[New Thread 0x7fffd3fff700 (LWP 19589)]
[New Thread 0x7fffda7fc700 (LWP 19590)]
[tcsetpgrp failed in terminal_inferior: 无效的参数]
[tcsetpgrp failed in terminal_inferior: 无效的参数]
[Thread 0x7fffd3fff700 (LWP 19589) exited]
[ WARN] [1673611745.896928552]: SPATIAL FILTER
[ WARN] [1673611745.897751122]: TEMPORAL FILTER
[ WARN] [1673611745.897761691]: SPECKLE FILTER
[ WARN] [1673611745.897780515]: class heritated, starting node...
[Thread 0x7fffdbfff700 (LWP 19588) exited]
[New Thread 0x7fffdb7fe700 (LWP 19601)]
[New Thread 0x7fffdaffd700 (LWP 19602)]
[New Thread 0x7fffd9ffb700 (LWP 19604)]
[New Thread 0x7fffd97fa700 (LWP 19605)]
[New Thread 0x7fffd8ff9700 (LWP 19606)]
[New Thread 0x7fffd37fe700 (LWP 19607)]
[New Thread 0x7fffd19fa700 (LWP 19608)]
[New Thread 0x7fffd11f9700 (LWP 19609)]
[New Thread 0x7fffd09f8700 (LWP 19610)]
[New Thread 0x7fffc941c700 (LWP 19611)]
[New Thread 0x7fffc8c1b700 (LWP 19612)]
[New Thread 0x7fffc841a700 (LWP 19613)]
[New Thread 0x7fffc7c19700 (LWP 19614)]
[Thread 0x7fffc7c19700 (LWP 19614) exited]
[Thread 0x7fffd9ffb700 (LWP 19604) exited]
[Thread 0x7fffc841a700 (LWP 19613) exited]
[Thread 0x7fffc8c1b700 (LWP 19612) exited]
[Thread 0x7fffc941c700 (LWP 19611) exited]
[Thread 0x7fffd09f8700 (LWP 19610) exited]
[Thread 0x7fffd11f9700 (LWP 19609) exited]
[Thread 0x7fffd19fa700 (LWP 19608) exited]
[Thread 0x7fffd37fe700 (LWP 19607) exited]
[Thread 0x7fffd8ff9700 (LWP 19606) exited]
[Thread 0x7fffdaffd700 (LWP 19602) exited]
[Thread 0x7fffdb7fe700 (LWP 19601) exited]
[Thread 0x7fffd97fa700 (LWP 19605) exited]
terminate called after throwing an instance of 'std::runtime_error'
  what():  IMU(0) - IMU invalid settings!: ROTATION_VECTOR output is unsupported. BMI270 supports only ACCELEROMETER_RAW and/or GYROSCOPE_RAW outputs.

Thread 1 "oakd_ros_node" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: 没有那个文件或目录.
(gdb)


Can you give me some advice? Thank u!

**Comment by Cxz-dev** [comment]
And when I type "bt", the log is:

#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff233e7f1 in __GI_abort () at abort.c:79
#2  0x00007ffff2993957 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff2999ae6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff2999b21 in std::terminate() ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff2999da9 in __cxa_rethrow ()
   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff2d40f3e in dai::DeviceBase::tryStartPipeline(dai::Pipeline const&) () from /usr/local/lib/libdepthai-core.so
#7  0x00007ffff2d32e5b in dai::Device::Device(dai::Pipeline const&) ()
   from /usr/local/lib/libdepthai-core.so
#8  0x00005555554133b8 in main (argc=<optimized out>, argv=0x7fffffffda98)
    at /home/cxz/dai_ws/src/oakd-ros-simple-main/src/main_ros.cpp:16

**Comment by ghost** [comment]
hello. i see the same error. ubuntu20, noetic, oak-d-pro w

**Comment by LanWu076** [comment]
anyone solved it? I have exactly the same error. no device is found.

**Comment by engcang** [comment]
@Cxz-dev , @LanWu076 , @asimonov-TII 

I also suffered the problem today and solved it by changing the USB3 cable.
I recommend to check `$ lsusb`, `$ dmesg -w`, and run `depthai_demo.py` from [official demo](https://github.com/luxonis/depthai) first. If you cannot run official demo, solve the cable and connection issue first. It is not the case of my code.

**Comment by ahjcahkl** [comment]
I think you can close the ROTATION_VECTOR in main_ros.cpp L189.I can delete the 'dai::IMUSensor::ROTATION_VECTOR' to close this function.

**Comment by engcang** [comment]
@ahjcahkl Hi.
Do you mean getting `dai::IMUSensor::ROTATION_VECTOR` occur the error?

**Comment by ahjcahkl** [comment]
I searched for relevant issues on Luxonis and found that someone has encountered the same problem as me (regarding BMI 270). According to the official explanation, some OAK-D-PRO models of the same model use BNO085's IMU, which supports 9-axis information. But BMI does not support it, which resulted in an error. I have reviewed the relevant code and found that if an error occurs, modifying the declaration during IMU initialization can solve the problem. The above are my thoughts.
